crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -718,7 +733,12 @@ impl UnableToDecryptReason {
     /// Returns true if this UTD is due to a missing room key (and hence might
     /// resolve itself if we wait a bit.)
     pub fn is_missing_room_key(&self) -> bool {
-        matches!(self, Self::MissingMegolmSession | Self::UnknownMegolmMessageIndex)
+        match self {
+            // Return true only if the key is not withheld on purpose.
+            Self::MissingMegolmSession(None) => true,
+            Self::UnknownMegolmMessageIndex => true,
+            _ => false,
+        }
```

Comment:
Surprised that the linter doesn't complain about this, but this should still be a `matches!` macro I think

Reply:
Not sure to get it, what do you suggest?

Reply:
```suggestion
        // Return true only if the key is not withheld on purpose.
        matches!(self, Self::MissingMegolmSession(None) | Self::UnknownMegolmMessageIndex)
```